File: api/src/main/java/me/slidingshadows/bookshelf/api/responses/BaseResponse.java
package me.slidingshadows.bookshelf.api.responses;

import java.util.ArrayList;
import java.util.List;

public class BaseResponse {
    private Boolean succeeded;
    private List<Error> errors;

    public BaseResponse() {
        this.succeeded = true;
        this.errors = new ArrayList<Error>();
    }

    public List<Error> getErrors() {
        return errors;
    }

    public Boolean getSucceeded() {
        return succeeded;
    }

    public void addError(String code, String description) {
        this.errors.add(new Error(code, description));
        this.succeeded = false;
    }
}

File: BookShelf.Logic/Responses/BaseResponse.cs
using BookShelf.Logic.Common;
using BookShelf.Logic.Models;
using FluentValidation.Results;
using System.Collections.Generic;

namespace BookShelf.Logic.Responses
{
    public class BaseResponse : IOperation
    {
        private List<ResponseError> _errors;

        public string RequestId { get; set; }
        public bool Succeeded { get; protected set; }
        public IEnumerable<ResponseError> Errors { get => _errors; }

        public BaseResponse()
        {
            Succeeded = true;
            _errors   = new List<ResponseError>();
        }

        public void Failed(string code, string description)
        {
            _errors.Add(new ResponseError()
            {
                Code        = code,
                Description = description,
            });

            Succeeded = false;
        }

        public bool FromResult(ValidationResult result)
        {
            if (!result.IsValid)
            {
                foreach (var faulure in result.Errors)
                {
                    Failed(faulure.ErrorCode, faulure.ErrorMessage);
                }
            }

            return Succeeded;
        }
    }
}
